x-pack/platform/plugins/shared/onechat/public/application/components/agents/edit/agent_form.tsx

```diff
@@ -125,16 +132,30 @@ export const AgentForm: React.FC<AgentFormProps> = ({ editingAgentId, onDelete }
     mode: 'onBlur',
     resolver: zodResolver(agentFormSchema),
   });
-  const { control, handleSubmit, reset, formState, watch } = formMethods;
+  const { control, handleSubmit, reset, formState, watch, getValues } = formMethods;
   const { errors, isDirty } = formState;
   const hasErrors = Object.keys(errors).length > 0;
 
+  useUnsavedChangesPrompt({
+    hasUnsavedChanges: isDirty,
+    history,
+    http,
+    navigateToUrl,
+    openConfirm,
+  });
+
   useEffect(() => {
     if (agentState && !isLoading) {
       reset(agentState);
     }
   }, [agentState, isLoading, reset]);
 
+  const handleCancel = useCallback(() => {
+    // Bypass unsaved changes prompt
+    reset(getValues());
```

Comment:
do we need resetting given that we are navigating away from this page anyway? (if page is opened again we will re-load data anyway)